File: 2551-apply-operations-to-an-array/2551-apply-operations-to-an-array.py
class Solution(object):
    def applyOperations(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        for i in range(len(nums)-1):
            if(nums[i]==nums[i+1]):
                nums[i]+=nums[i+1]
                nums[i+1]=0
        j=0
        for i in range(len(nums)):
            if(nums[i]!=0):
                nums[j]=nums[i]
                j+=1
        for i in range(j,len(nums)):
            nums[i]=0
        return nums

        

File: 2551-apply-operations-to-an-array/2551-apply-operations-to-an-array.cpp
class Solution {
public:
    vector<int> applyOperations(vector<int>& nums) {
        for (int i=0;i<nums.size()-1;i++){
            if(nums[i]==nums[i+1]){
                nums[i]+=nums[i+1];
                nums[i+1]=0;
            }

        }
        int j=0;
        for (int i=0;i<nums.size();i++){
            if(nums[i]!=0){
                nums[j]=nums[i];
                j+=1;
            }

        }
        while(j<nums.size()){
            nums[j]=0;
            j++;
        }
        return nums;
    }
};